src/NuGet.Core/NuGet.PackageManagement/Utility/PackagesConfigLockFileUtility.cs

```diff
@@ -0,0 +1,158 @@
+                }
+            }
+        }
+
+        internal static string GetPackagesLockFilePath(MSBuildNuGetProject msbuildProject)
+        {
+            var directory = Path.GetDirectoryName(msbuildProject.MSBuildProjectPath);
+            var msbuildProperty = msbuildProject?.ProjectSystem?.GetPropertyValue("NuGetLockFilePath");
+            if (msbuildProperty is string nuGetLockFilePathValue)
+            {
+                if (!string.IsNullOrWhiteSpace(nuGetLockFilePathValue))
+                {
+                    return Path.Combine(directory, nuGetLockFilePathValue);
+                }
+            }
+
+            return PackagesLockFileUtilities.GetNuGetLockFilePath(directory, Path.GetFileNameWithoutExtension(msbuildProject.MSBuildProjectPath));
+        }
+
+        private static bool IsPackagesLockFileExcplicitlyEnabled(MSBuildNuGetProject msbuildProject)
+        {
+            var msbuildProperty = msbuildProject?.ProjectSystem?.GetPropertyValue("RestorePackagesWithLockFile");
+            if (msbuildProperty is string restorePackagesWithLockFileValue)
+            {
+                bool.TryParse(restorePackagesWithLockFileValue, out var useLockFile);
+                return useLockFile;
+            }
+
+            return false;
+        }
+
+        internal static PackagesLockFile GetLockFile(bool lockFileExists, string lockFileName)
+        {
+            PackagesLockFile lockFile;
+
+            if (lockFileExists)
+            {
+                lockFile = PackagesLockFileFormat.Read(lockFileName);
+            }
+            else
+            {
+                lockFile = new PackagesLockFile();
+                lockFile.Targets.Add(new PackagesLockFileTarget
+                {
+                    TargetFramework = new NuGetFramework(FrameworkConstants.SpecialIdentifiers.Any)
+                });
+            }
+
+            return lockFile;
+        }
+
+        internal static async Task ApplyChangesAsync(
+            PackagesLockFile lockFile,
+            List<NuGetProjectAction> actionsList,
+            IContentHashUtility contentHashUtil,
+            CancellationToken token)
+        {
+            RemoveUninstalledPackages(lockFile,
+                actionsList.Where(a => a.NuGetProjectActionType == NuGetProjectActionType.Uninstall));
+            await AddInstalledPackagesAsync(lockFile,
+                actionsList.Where(a => a.NuGetProjectActionType == NuGetProjectActionType.Install),
+                contentHashUtil,
+                token);
+        }
+
+        private static void RemoveUninstalledPackages(PackagesLockFile lockFile, IEnumerable<NuGetProjectAction> actionsList)
+        {
+            foreach (var toUninstall in actionsList)
+            {
+                Debug.Assert(toUninstall.NuGetProjectActionType == NuGetProjectActionType.Uninstall);
+
+                foreach (var installed in lockFile.Targets[0].Dependencies)
```

Comment:
PC lock files will always use the "Any,V0.0" TFM. Partly because it's easier (the actions list doesn't specify which TFM the package should use, so we would have to get additional information from elsewhere to use the 'actual' TFM). Partly because PC projects can't multi-target and doesn't have the capability to have conditional references depending on target framework, like PR can. The PC file is a flat a flat list, so the lock file can be too.